app/services/LocationService.js

```diff
@@ -195,10 +203,39 @@ export default class LocationServices {
             console.log('[INFO] BackgroundGeolocation services enabled', status.locationServicesEnabled);
             console.log('[INFO] BackgroundGeolocation auth status: ' + status.authorization);
 
-            // you don't need to check status before start (this is just the example)
-            if (!status.isRunning) {
-                BackgroundGeolocation.start(); //triggers start on start event
+            BackgroundGeolocation.start(); //triggers start on start event
+
+            if (!status.locationServicesEnabled) {
+                // we need to set delay or otherwise alert may not be shown
+                setTimeout(() =>
+                    Alert.alert('Private Kit requires location services to be enabled', 'Would you like to open location settings?', [{
+                        text: 'Yes',
+                        onPress: () => BackgroundGeolocation.showLocationSettings()
+                    },
+                    {
+                        text: 'No',
+                        onPress: () => console.log('No Pressed'),
+                        style: 'cancel'
+                    }
+                    ]), 1000);
+            }
+            else if (!status.authorization) {
```

Comment:
If I don't authorize and press Yes for settings, then I end up at App settings which doesn't have location information anywhere.